File: functions/update_budget.c
#include <stdio.h>

float input(char type[]) {
    float var;
    printf("Monthly %s: ", type);
    scanf("%f", &var);
    return var;
}

void percent(char type[], float amount, float income) {
    if (income > 0) {
        int per = (amount / income) * 100;
        printf("Your %s is %d%% of your income.\n", type, per);
    } else {
        printf("Income must be greater than 0 to calculate percentage.\n");
    }
}

int main(void) {
    printf("This is a budget calculator.\n");
    
    float income = input("income");
    float rent = input("rent");
    float utilities = input("utilities");
    float groceries = input("groceries");
    float transportation = input("transportation");

    float savings = income * 0.2;
    float expenses = rent + utilities + groceries + transportation;
    float spend = income - expenses - savings;

    printf("You make $%.2f\n", income);
    printf("Your expenses are $%.2f\n", expenses);
    printf("Your savings are $%.2f\n", savings);
    printf("Your spending money is $%.2f\n", spend);

    percent("rent", rent, income);
    percent("utilities", utilities, income);
    percent("groceries", groceries, income);
    percent("transportation", transportation, income);
    percent("expenses", expenses, income);
    percent("savings", savings, income);
    percent("spending money", spend, income);

    return 0;
}

File: functions/update_budget.py
print("This is a budget calculator!\n")


income = float(input("How much do you make a month: "))
rent = float(input("How much does your rent cost: "))
utilities = float(input("How much do your utilities cost: "))
groceries = float(input("How much do your groceries cost: "))
transportation = float(input("How much does your transportation cost: "))


savings = income * 0.2
expenses = rent + utilities + groceries + transportation
spending = income - expenses - savings


rent_percentage = (rent / income) * 100
utilities_percentage = (utilities / income) * 100
groceries_percentage = (groceries / income) * 100
transportation_percentage = (transportation / income) * 100
savings_percentage = (savings / income) * 100
expenses_percentage = (expenses / income) * 100


print(f"\nYour monthly income is ${income:.2f}")
print(f"Your monthly expenses are ${expenses:.2f}")
print(f"Your monthly savings are ${savings:.2f}")
print(f"Your monthly spending money is ${spending:.2f}")
print(f"Your rent is {rent_percentage:.2f}% of your income.")
print(f"Your utilities are {utilities_percentage:.2f}% of your income.")
print(f"Your groceries are {groceries_percentage:.2f}% of your income.")
print(f"Your transportation is {transportation_percentage:.2f}% of your income.")
print(f"Your savings are {savings_percentage:.2f}% of your income.")
print(f"Your total expenses are {expenses_percentage:.2f}% of your income.")